docs/v3.x/getting-started/react.md

```diff
@@ -0,0 +1,577 @@
+    if (error) {
+       return <div>An error occured: {error.message}</div>
+    }
+    return (
+      <div className="App">
+        <ul>
+          {this.state.restaurants.map(restaurant => <li key={restaurant.id}>{restaurant.name}</li>)}
+        </ul>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+:::
+
+::: tab fetch
+
+`./src/App.js`
+
+```js
+import React from 'react';
+
+class App extends React.Component {
+
+  state = {
+    restaurants: [],
+    error: null
+  }
+
+  componentDidMount() {
+    fetch("http://localhost:1337/restaurants", {
+      headers: {
+         'Content-Type': 'application/json'
+      },
+    }).then(response => response.json())
+      .then(data => {
+        const restaurants = data;
+        this.setState({ restaurants });
+      })
+      .catch(error => {
+        this.setState({ error });
+      });
+  }
+
+  render() {
+    const { error, restaurant } = this.state
+    if (error) {
+       return <div>An error occured: {error.message}</div>
+    }
+    return (
+      <div className="App">
+        <ul>
+          { this.state.restaurants.map(restaurant => <li key={restaurant.id}>{restaurant.name}</li>)}
+        </ul>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+:::
+
+::::
+
+### POST Request your collection type
+
+Execute a POST request on the `restaurant` Collection Type in order to create a restaurant.
+
+Be sure that you activated the `create` permission for the `restaurant` Collection Type and the `find` permission fot the `category` Collection type.
+
+:::: tabs
+
+::: tab axios
+
+*Request*
+
+```js
+axios.post('http://localhost:1337/restaurants', {
+    name: 'Dolemon Sushi',
+    description: 'Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious'
+  })
+  .then(response => {
+    console.log(response);
+  })
+  .catch(error => {
+    console.log(error);
+  });
+```
+
+:::
+
+::: tab fetch
+
+*Request*
+
+```js
+fetch('http://localhost:1337/restaurants', {
+   method: 'POST',
+   headers: {
+      'Content-Type': 'application/json'
+   },
+   body: JSON.stringify({
+     name: 'Dolemon Sushi',
+     description: 'Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious'
+   })
+ }).then(response => {
+   return response.json();
+ }).then(data => {
+   console.log(data);
+ }).catch(error => {
+   console.error(error);
+ });
+```
+
+:::
+
+*Response*
+
+```json
+{
+    "id": 2,
+    "name": "Dolemon Sushi",
+    "description": "Unmissable Japanese Sushi restaurant. The cheese and salmon makis are delicious",
+    "created_by": null,
+    "updated_by": null,
+    "created_at": "2020-08-04T09:57:11.669Z",
+    "updated_at": "2020-08-04T09:57:11.669Z",
+    "categories": [
+    ]
+}
+```
+
+::::
+
+### Example
+
+:::: tabs
+
+::: tab axios
+
+`./src/App.js`
+
+```js
+import React from 'react';
+import axios from 'axios'
+
+class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
You should separate the objects of your state to have a special `modifiedData` object to handle your form state.

```
this.state = {
 modifiedData: {
        name: '',
        description: '',
        categories: [] // If it's plural it needs to be an array
      },
      allCategories: [],
      error: null
}
```